python/ray/train/_internal/dataset_spec.py

```diff
@@ -196,10 +204,12 @@ def get_dataset_shards(
                 # cluster hot-spots since we already randomized the based blocks, but
                 # can help with improving randomness in combination with local shuffle.
                 if config.randomize_block_order and not config.global_shuffle:
+                    # TODO(swang): Should randomize block order across the
+                    # original dataset, not the window.
                     dataset = dataset.randomize_block_order_each_window()
 
             if config.global_shuffle:
-                if config.use_stream_api:
+                if config.max_object_store_memory_fraction >= 0:
                     dataset = dataset.random_shuffle_each_window()
                 else:
```

Comment:
Nit: Do we perform any input validation on `max_object_store_memory_fraction`? If not, we could go through this code path if `max_object_store_memory_fraction` is an invalid value like `max_object_store_memory_fraction=True`.